app/src/main/java/com/architectcoders/arquitectomarvel/data/local/RoomDataSource.kt

```diff
@@ -0,0 +1,32 @@
+package com.architectcoders.arquitectomarvel.data.local
+
+import com.architectcoders.arquitectomarvel.data.mappers.toCharacterEntity
+import com.architectcoders.arquitectomarvel.data.mappers.toDetailedComicEntity
+import com.architectcoders.module.data.sources.LocalDataSource
+import com.architectcoders.module.domain.local_models.DetailedComic
+import com.architectcoders.module.domain.remote_models.Characters.Result
+
+class RoomDataSource(resultDatabase: ResultDatabase) : LocalDataSource {
+
+    private val dao: ResultDao = resultDatabase.resultDao
+
+    override suspend fun insertFavoriteCharacter(toCharacterEntity: Result) {
+        dao.insertFavoriteCharacter(toCharacterEntity.toCharacterEntity())
```

Comment:
@joseantoniocampillo is the coroutine context type Dispatchers.IO?

Reply:
The database work is guaranteed to be run on a non-UI Dispatcher with Room.

Reply:
@joseantoniocampillo that function is executed on the main thread (viewModelScope), so we need to change the dispacher to the Dispatcher.IO